mama/c_cpp/src/c/reservedfields.c

```diff
@@ -114,7 +111,7 @@ void initReservedFields (void)
 {
     CREATE_FIELD (MsgType,            U8);
     CREATE_FIELD (MsgStatus,          U8);
-    CREATE_FIELD (FieldIndex,         U16);
+    CREATE_FIELD (FieldIndex,         VECTOR_U16);
```

Comment:
In theory `MdFieldIndex` is a vector of field IDs - it was formerly used in ticker plants to store a list of IDs contained in the rest of the update message (or in sub messages for the same purpose). I'm not aware of any publishers currently using it however. 

That said, `VECTOR_U16` was the original format for the field before the changes 83fc5dfd15. If there are any legacy publishers using this field out there, the current format breaks both them and consumers. Since the change is to correct a mistake made before, I think it should be allowed.